components/scream/src/mct_coupling/scream_scorpio_interface.F90

```diff
@@ -630,9 +687,16 @@ subroutine get_pio_atm_file(filename,pio_atm_file,found)
     allocate(curr%pio_file)
     ! Create and initialize the new pio file:
     curr%pio_file%filename = trim(filename)
-    curr%pio_file%numRecs = 0
-    call eam_pio_createfile(curr%pio_file%pioFileDesc,trim(curr%pio_file%filename))
-    pio_atm_file => curr%pio_file  
+    if (purpose_loc == 0) then
+      curr%pio_file%numRecs = 0
+      call eam_pio_createfile(curr%pio_file%pioFileDesc,trim(curr%pio_file%filename))
+    elseif (purpose_loc == 1) then
+      call eam_pio_openfile(curr%pio_file%pioFileDesc,trim(curr%pio_file%filename))
+      found = .true.
```

Comment:
This is where my F90 skills betray me: shouldn't this assignment be guarded by a `if(present(found))` clause?

Reply:
I think you are correct.  I guess I didn't have any cases where I was not passing the optional argument of `found`